components/web_discovery/browser/background_credential_helper.h

```diff
@@ -0,0 +1,61 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_WEB_DISCOVERY_BROWSER_BACKGROUND_CREDENTIAL_HELPER_H_
+#define BRAVE_COMPONENTS_WEB_DISCOVERY_BROWSER_BACKGROUND_CREDENTIAL_HELPER_H_
+
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include "brave/components/web_discovery/browser/rsa.h"
+#include "crypto/rsa_private_key.h"
+
+namespace web_discovery {
+
+struct GenerateJoinRequestResult {
+  GenerateJoinRequestResult(std::string join_request_b64,
+                            std::vector<uint8_t> join_gsk,
+                            std::string signature);
+
+  ~GenerateJoinRequestResult();
+
+  GenerateJoinRequestResult(const GenerateJoinRequestResult&);
+  GenerateJoinRequestResult& operator=(const GenerateJoinRequestResult&);
+
+  std::string join_request_b64;
+  std::vector<uint8_t> join_gsk;
+  std::string signature;
+};
+
+class BackgroundCredentialHelper {
+ public:
+  static std::unique_ptr<BackgroundCredentialHelper> Create();
+
+  virtual ~BackgroundCredentialHelper() = default;
+
+  virtual void UseFixedSeedForTesting() = 0;
+
+  virtual std::unique_ptr<EncodedRSAKeyPair> GenerateRSAKey() = 0;
```

Comment:
it's a bit confusing to me that there's both `GenerateRSAKey` and `SetRSAKey` on `BackgroundCredentialHelper`. I think it would make more sense if `GenerateRSAKey` was in `CredentialManager` and also `GenerateRSAKey` -> `GenerateEncodedRSAKeyPair`